analysis_notebooks/Attribute Analysis.ipynb

```diff
@@ -981,6 +984,296 @@
+    "        for k in range(num_attrs):\n",
+    "            b_dist.extend([(chunk[0] / np.sqrt(chunk[1]*chunk[2])) for chunk in distances[i][k] if chunk[2] != 0 and chunk[1] != 0 and j != k])\n",
+    "            b_file.extend([chunk[3] for chunk in distances[i][k] if chunk[2] != 0 and chunk[1] != 0 and j != k])\n",
+    "            b_index.extend([chunk[4] for chunk in distances[i][k] if chunk[2] != 0 and chunk[1] != 0 and j != k])\n",
+    "        \n",
+    "        t, p = stats.ttest_ind(a_dist, b_dist, equal_var=False)\n",
+    "\n",
+    "        cat_p_values.append(p)\n",
+    "        cat_distr_dists.append([a_dist, b_dist])\n",
+    "        cat_dists.append([np.mean(a_dist), np.mean(b_dist)])\n",
+    "        cat_file_paths.append([a_file, b_file])\n",
+    "        cat_indices.append([a_index, b_index])\n",
+    "\n",
+    "        a_mean = np.mean(a_dist)\n",
+    "        b_mean = np.mean(b_dist)\n",
+    "        if not np.isnan(a_mean) and not np.isnan(b_mean) and not len(a_dist) < 5 and not len(b_dist) < 5:\n",
+    "            if (a_mean >= b_mean and a_mean/b_mean > largest_ratio and p < 0.05):\n",
+    "                largest_ratio = a_mean/b_mean\n",
+    "                ratio_idx = j\n",
+    "            if (a_mean < b_mean and b_mean/a_mean > largest_ratio and p < 0.05):\n",
+    "                largest_ratio = b_mean/a_mean\n",
+    "                ratio_idx = j\n",
+    "\n",
+    "    if ratio_idx != -1:\n",
+    "        attr_dists.append(cat_dists[ratio_idx])\n",
+    "        attr_distr_dists.append(cat_distr_dists[ratio_idx])\n",
+    "        attr_file_paths.append(cat_file_paths[ratio_idx])\n",
+    "        attr_indices.append(cat_indices[ratio_idx])\n",
+    "        p_values.append(cat_p_values[ratio_idx])\n",
+    "        ratio_idxs.append(ratio_idx)\n",
+    "        xaxis.append(names[categories[i]])\n",
+    "\n",
+    "# Only retaining inforamtion for distances that are statistically significantly different, and are not people\n",
+    "indices_to_keep = [i for i in range(len(attr_dists)) if categories[i] not in dataset.people_labels]\n",
+    "attr_sig_dists = np.array(attr_dists)[indices_to_keep]\n",
+    "attr_distr_dists = np.array(attr_distr_dists)[indices_to_keep]\n",
+    "attr_file_paths = np.array(attr_file_paths)[indices_to_keep]\n",
+    "attr_indices = np.array(attr_indices)[indices_to_keep]\n",
+    "xaxis_sig = np.array(xaxis)[indices_to_keep]\n",
+    "pvalues_sig = np.array(p_values)[indices_to_keep]\n",
+    "ratio_idxs = np.array(ratio_idxs)[indices_to_keep]\n",
+    "\n",
+    "all_ratios = []\n",
+    "for i in range(len(attr_sig_dists)):\n",
+    "    in_count = attr_sig_dists[i][0]\n",
+    "    out_count = attr_sig_dists[i][1]\n",
+    "    if in_count > out_count:\n",
+    "        all_ratios.append(in_count/out_count)\n",
+    "    else:\n",
+    "        all_ratios.append(out_count/-in_count)\n",
+    "    if np.absolute(all_ratios[-1]) <= 1:\n",
```

Comment:
what's this here for? if the ratio is 1?

Reply:
Is this referencing line 1096? That's also there in the gender analysis notebook, I assumed it was there to prune the ratios that were small, but now that I think about it it probably isn't needed.